### ISSUE TITLE
go/build: Default does not honor `go env -w GOPATH`

### ISSUE THREAD

**Issue by heschik** [golang/go#39838]
`go/build` has no logic for reading the `GOENV` file, so it will ignore any `GOPATH` setting there.
https://github.com/golang/go/blob/7eb5941b95a588a23f18fa4c22fe42ff0119c311/src/go/build/build.go#L275-L292

**Comment by mvdan** [comment]
Since `build.Default` is a global variable, wouldn't a call to `go env` add quite a lot of cost to the init function?

**Comment by heschik** [comment]
Definitely, 20ms at least. I was assuming that it would directly open and parse the config file.

**Comment by mvdan** [comment]
Gotcha - I assume that is acceptable, given that `go/build` and `cmd/go` are released together.
